clang/lib/CIR/Dialect/IR/UnimplementedFeatureGuarding.h

```diff
@@ -0,0 +1,27 @@
+//===---- UnimplementedFeatureGuarding.h - Checks against NYI ---*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file introduces some helper classes to guard against features that
+// CIR dialect supports that we do not have and also do not have great ways to
+// assert against.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_CIR_DIALECT_IR_UFG
+#define LLVM_CLANG_LIB_CIR_DIALECT_IR_UFG
+
+namespace cir {
+
+struct CIRDialectUnimplementedFeature {
```

Comment:
Rename `CIRDialectUnimplementedFeature` -> `MissingFeatures`. CIRGen is a bit different because we unimplemented tracks things relative to traditional codegen. Here we don't have a dialect to compare against, so "missing" sounds more accurate.

Reply:
Also rename the file to "MissingFeatures.h"

Reply:
Updated.